core-library/src/main/java/org/silverpeas/core/cmis/model/CmisObject.java

```diff
@@ -0,0 +1,358 @@
+   */
+  public String getName() {
+    return name;
+  }
+
+  /**
+   * Gets the language in which some of the object's attributes are localized.
+   * @return an ISO 639-1 language code corresponding to the language of the user behind the
+   * request.
+   */
+  public String getLanguage() {
+    return language;
+  }
+
+  /**
+   * Gets the identifier of the base type from which the type of the CMIS object is derived.
+   * @returna a {@link BaseTypeId} enumeration value.
+   */
+  @Override
+  public abstract BaseTypeId getBaseTypeId();
+
+  /**
+   * Gets the identifier of the type of this CMIS object.
+   * @return a {@link TypeId} enumeration value.
+   */
+  public abstract TypeId getTypeId();
+
+  /**
+   * Gets a description of this CMIS object.
+   * @return a short description about it.
+   */
+  public String getDescription() {
+    return description;
+  }
+
+  /**
+   * Sets a description about this CMIS object.
+   * @param description a short text to set.
+   * @param <T> the concrete Java type of this CMIS object.
+   * @return itself.
+   */
+  public <T extends CmisObject> T setDescription(final String description) {
+    this.description = description;
+    return self();
+  }
+
+  /**
+   * Gets the full name of the user that has authored this CMIS object.
+   * @return the full name of the creator.
+   */
+  public String getCreator() {
+    return creator;
+  }
+
+  /**
+   * Sets the full name of the user that has created this CMIS object.
+   * @param creator the full name of a user in Silverpeas.
+   * @param <T> the concrete Java type of this CMIS object.
+   * @return itself.
+   */
+  public <T extends CmisObject> T setCreator(final String creator) {
+    this.creator = creator;
+    return self();
+  }
+
+  /**
+   * Gets the full name of the user that has lastly modified this CMIS object. By default, the first
+   * modifier is the creator of this object.
+   * @return the full name of the last modifier.
+   */
+  public String getLastModifier() {
+    return lastModifier;
+  }
+
+  /**
+   * Sets the full name of the user that has modified this CMIS object.
+   * @param lastModifier the full name of a user in Silverpeas.
+   * @param <T> the concrete Java type of this CMIS object.
+   * @return itself.
+   */
+  public <T extends CmisObject> T setLastModifier(final String lastModifier) {
+    this.lastModifier = lastModifier;
+    return self();
+  }
+
+  /**
+   * Gets the date at which this CMIS object was created.
+   * @return the number of milliseconds since the EPOCH January 1, 1970, 00:00:00 GMT
+   */
+  public long getCreationDate() {
+    return creationDate;
+  }
+
+  /**
+   * Sets the date at which this CMIS object was created.
+   * @param creationDate a number of milliseconds since the EPOCH January 1, 1970, 00:00:00 GMT
+   * @param <T> the concrete Java type of this CMIS object.
+   * @return itself.
+   */
+  public <T extends CmisObject> T setCreationDate(final long creationDate) {
```

Comment:
In CMIS, the date are expressed as the number of milliseconds since the EPOCH January 1, 1970, 00:00:00 GMT. OpenCMIS waits for such representation of the dates and this is why, as _CmisObject_ represents an object exposed in a CMIS repository, the dates here are represented as such.